File: sem3/map/Lab/3/toylanguage/src/main/java/com/company/Models/Statements/PrintStatement.java
package com.company.Models.Statements;

import com.company.Models.*;
import com.company.Models.ProgramState.ProgramState;
import com.company.Utils.MyIDictionary;
import com.company.Utils.MyIList;

public class PrintStatement implements IStatement {
    private Expression expression;

    public PrintStatement(Expression expression) {
        this.expression = expression;
    }

    @Override
    public ProgramState execute(ProgramState state) throws Exception {
        MyIList<Integer> output = state.getOutput();
        MyIDictionary<String, Integer> symTable = state.getSymTable();

        int expressionOutput = this.expression.eval(symTable);

        System.out.println(expressionOutput);
        output.add(expressionOutput);

        return state;
    }

    @Override
    public String toString() {
        return "print (" + this.expression.toString() + ")";
    }
}

File: sem3/map/Lab/toylanguage_C#/ToyLanguage/ToyLanguage/Models/Statements/PrintStatement.cs
using System;
using ToyLanguage.Models.Interfaces;
using ToyLanguage.Models.States;

namespace ToyLanguage.Models.Statements
{
    public class PrintStatement : IStatement
    {
        private readonly IExpression _expression;

        public PrintStatement(IExpression expression)
        {
            _expression = expression;
        }
        
        public ProgramState Execute(ProgramState state)
        {
            var expressionValue = _expression.Eval(state);
            
            state.Output.Add(expressionValue.ToString());
            
            Console.WriteLine(expressionValue);
            
            return state;
        }

        public override string ToString()
        {
            return $"Print ({_expression})";
        }
    }
}